ksql-engine/src/test/java/io/confluent/ksql/integration/StreamsSelectAndProjectIntTest.java

```diff
@@ -30,22 +30,28 @@
 
   private IntegrationTestHarness testHarness;
   private KsqlContext ksqlContext;
-  private Map<String, RecordMetadata> recordMetadataMap;
-  private final String topicName = "TestTopic";
+  private Map<String, RecordMetadata> jsonRecordMetadataMap;
+  private final String jsonTopicName = "jsonTopic";
+  private final String jsonStreamName = "orders_json";
+  private Map<String, RecordMetadata> avroRecordMetadataMap;
+  private final String avroTopicName = "avroTopic";
+  private final String avroStreamName = "orders_avro";
   private OrderDataProvider dataProvider;
 
   @Before
   public void before() throws Exception {
-    testHarness = new IntegrationTestHarness(DataSource.DataSourceSerDe.JSON.name());
+    testHarness = new IntegrationTestHarness();
```

Comment:
Thanks for the patch! It's a step in the right direction. However, I think we can drop even more duplicate code by using the `Parameterized` construct of Junit: https://github.com/junit-team/junit4/wiki/Parameterized-tests

This way, we can have have single test cases like `testSelectProject`, `testSelectStar`, `testSelectFilter`, and run them with both Avro and JSON without duplicating the code. It looks like the tests are identical except for the input stream, so it should be fairly straightforward to parameterize.

Reply:
I parameterized the tests but didn't use the `Parametrized` since there are delimited tests that were not the same as the other ones.